### ISSUE TITLE
controlgroup with checkbox: cannot remove

### ISSUE THREAD

**Issue by rejetto** [jquery/jquery-mobile#8119]
expected: the checkbox is removed
what happens: an exception is thrown

to reproduce
1. go to http://demos.jquerymobile.com/1.4.5/controlgroup-dynamic/
2. click checkbox
3. click append
4. click "Checkbox 1"

**Comment by kapilgarg1996** [comment]
@arschmitz the error is thrown due to <code>this._getInputSet().checkboxradio("refresh")</code>
Shouldn't we change it to <code>this.refresh()</code>

**Comment by kapilgarg1996** [comment]
@jaspermdegroot It is duplicate of #6087.
I think we should update the demos page using that solution which you suggested on #6087

**Comment by jaspermdegroot** [comment]
@kapilgarg1996 

Yeah it's the same problem, good catch. I am not sure if we should apply that solution though.

The root of the problem is that our demo tries to do something (removing a checkbox when you click on it) that shouldn't want to do in a real use case. If we need to add a ugly workaround just to make our demo work I think it proofs that it's not a good demo. We need to update it anyway for 1.5 so I think it's better to look into a different UI for removing and hiding elements in the controlgroup.

**Comment by kapilgarg1996** [comment]
@jaspermdegroot So no need to look into this issue now ?

**Comment by rejetto** [comment]
@jaspermdegroot 
what if you are removing it because you want to move it to a different group (because of the ticking).
Is it a real use case? That's how i stumbled on it.
I'm not saying it for me, as i already workarounded it (by moving the element) but it is better if people don't meet this problem or they will think it's the lib's fault.

**Comment by jaspermdegroot** [comment]
@kapilgarg1996 - We can look into it now, but in the 1.5-dev branch, not on master.

@rejetto - Thanks for reporting the issue.
I only meant removing the checkbox when you click on it, so you won't get good feedback (don't see anymore which checkbox you checked) and can't uncheck it. That's a very bad UI.
Moving it from one controlgroup to another (to group unchecked and checked items separately) is fine of course. But in that case it's best to do what you did, moving the enhanced element.
I think it's better to teach that than to try to fix this in the framework (if possible) or to show an ugly workaround with ```setTimeout```.

**Comment by arschmitz** [comment]
This is all kind of a moot point. Controlgroup is already replaced in 1.5 with a whole new widget. Any issues need to be tested on that widget and if applicable opened on the UI repo once it lands on master or reported to the ui wiki http://wiki.jqueryui.com/w/page/71756381/Button until then. I'm going to closes this @rejetto feel free to test this with the 1.5-dev branch and if you have issues with that report them in a new issue
